sdk/src/config.rs

```diff
@@ -105,7 +104,7 @@ pub struct GadgetConfiguration<RwLock: lock_api::RawRwLock> {
     /// The Port of the Network that will be interacted with
     pub bind_port: u16,
     /// The Address of the Network that will be interacted with
-    pub bind_addr: IpAddr,
+    pub bind_addr: String,
```

Comment:
Does this need to be a string? Using `core::net` types would be better if possible.